pkg/spi/resource/spi.go

```diff
@@ -0,0 +1,49 @@
+package resource
+
+import (
+	"encoding/json"
+
+	"github.com/docker/infrakit/pkg/spi"
+)
+
+// InterfaceSpec is the current name and version of the resource API.
+var InterfaceSpec = spi.InterfaceSpec{
+	Name:    "Resource",
+	Version: "0.1.0",
+}
+
+// Plugin is a vendor-agnostic API used to create and manage resources with an infrastructure provider.
+type Plugin interface {
```

Comment:
This is virtually identical to the `instance` spi, except for the addition of a `resourceType` in some of methods.

That you had to duplicate code tells me there's an issue with our current model.  Perhaps we should visit the SPI of instance itself?   

What if we can do something to the instance spi itself or change the way a plugin exposes the RPC endpoints, etc. so that all the different resource types all implement the same spi -- compute, networks, volumes, etc., that would be nice.   Let me investigate this.  Thoughts?